### PR TITLE
Reconstruct the box tree incrementally in some case

### PR DESCRIPTION
**Pull request by nox**
Nothing to see here for now. I just want to do a try run.

### CODE REVIEW

**Comment by nox** [comment]
```
  ▶ Unexpected subtest result in /css/cssom-view/elementFromPoint.html:
  │ FAIL [expected PASS] First element to get mouse events with pointer-events css
  │   → assert_equals: Should have returned the element with id `yellow` as element with `teal` has `pointer-events:none` expected Element node <div id="yellow" class="size yellow">&nbsp;</div> but got Element node <html><head><title>cssom-view - elementFromPoint</title>
  │   → ...
  │ 
  │ window.onload/<@http://web-platform.test:8000/css/cssom-view/elementFromPoint.html:113:28
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1977:25
  │ test@http://web-platform.test:8000/resources/testharness.js:535:30
  └ window.onload@http://web-platform.test:8000/css/cssom-view/elementFromPoint.html:112:11
```

Only one remaining failure, that's nice and unexpected.

**Review by SimonSapin — APPROVED** [review]
If I’m reading this correctly, the high-level algorithm is: if all DOM changes are contained within descendants of an abspos element, recontruct that element and its sub-tree. Otherwise reconstruct the entire document. Is this right?

Why only abspos? Could block formatting context (or independent formatting context) not be a sufficient boundary?

**Comment by SimonSapin** [comment]
(`rustfmt` is unhappy)

File: components/layout_2020/flow/root.rs [hunk 1]
```diff
@@ -88,6 +91,121 @@ impl BoxTree {
...
+                                inline_level_box.clone(),
+                            )
+                        },
+                        _ => return None,
+                    },
+                };
+            Some((primary_style.clone(), display_inside, update_point))
+        }
+
+        loop {
```
[the remark this answers is not in the document]
Comment (SimonSapin) [reply components/layout_2020/flow/root.rs:195]: As discussed on Matrix, this iterator is not available on wrapper types for DOM types in layout. They could be added, but no need to block this PR on that.

**Pull request merged**